### DISCUSSION TITLE
Clean builds with clang?

### DISCUSSION THREAD

**Discussion by cconvey** [campreilly/UnderSeaModelingLibrary#275] [Ideas]
How interested are we in having warning-free builds when using clang++?

I'm using clang-17, and I'm getting a lot (18 occurrences) of compilation noise like this:
```
 98%] Linking CXX executable reverb_analytic_test
[ 98%] Built target reverb_analytic_test
In file included from /home/cconvey/r/cconvey/usml/usml/sensors/test/sensors_test.cc:9:
/home/cconvey/r/cconvey/usml/usml/../usml/eigenverbs/eigenverb_collection.h:13:32: warning: unknown warning group '-Wmaybe-uninitialized', ignored [-Wunknown-warning-option]
   13 | #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
      |                                ^
1 warning generated.
In file included from /home/cconvey/r/cconvey/usml/usml/waveq3d/test/reverb_test.cc:7:
/home/cconvey/r/cconvey/usml/usml/../usml/eigenverbs/eigenverb_collection.h:13:32: warning: unknown warning group '-Wmaybe-uninitialized', ignored [-Wunknown-warning-option]
   13 | #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
      |                                ^
1 warning generated.
```

I can write a fix for this, but I'll only make it pretty if there's a chance of it getting upstreamed into this repo.

**Comment by campreilly** [comment]
I want to see all warnings fixed, wherever practical. Cases that I don’t think of as practical are things like forbidding “magic numbers” in the body of a unit test. And I plan to accept all changes. Sean Reilly401-523-5780 (cell)On Jul 4, 2024, at 12:01 PM, Christian Convey ***@***.***> wrote:﻿
How interested are we in having warning-free builds when using clang++?
I'm using clang-17, and I'm getting a lot (18 occurrences) of compilation noise like this:
 98%] Linking CXX executable reverb_analytic_test
[ 98%] Built target reverb_analytic_test
In file included from /home/cconvey/r/cconvey/usml/usml/sensors/test/sensors_test.cc:9:
/home/cconvey/r/cconvey/usml/usml/../usml/eigenverbs/eigenverb_collection.h:13:32: warning: unknown warning group '-Wmaybe-uninitialized', ignored [-Wunknown-warning-option]
   13 | #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
      |                                ^
1 warning generated.
In file included from /home/cconvey/r/cconvey/usml/usml/waveq3d/test/reverb_test.cc:7:
/home/cconvey/r/cconvey/usml/usml/../usml/eigenverbs/eigenverb_collection.h:13:32: warning: unknown warning group '-Wmaybe-uninitialized', ignored [-Wunknown-warning-option]
   13 | #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
      |                                ^
1 warning generated.

I can write a fix for this, but I'll only make it pretty if there's a chance of it getting upstreamed into this repo.

**Comment by cconvey** [comment]
Heads up, I just enabled "-Wall" with clang-17.  Bug reports and/or PRs incoming.
